server/src/main/java/com/thoughtworks/go/server/service/BuildAssignmentService.java

```diff
@@ -314,15 +315,15 @@ private Work createWork(final AgentInstance agent, final JobPlan job) {
         }
     }
 
-    private EnvironmentVariableContext getEnvironmentVariableContextFromEnvironment(String pipelineName) {
+    EnvironmentVariableContext getEnvironmentVariableContextFromEnvironment(String pipelineName) {
         EnvironmentConfig environmentConfig = environmentConfigService.environmentForPipeline(pipelineName);
```

Comment:
Optional: 
Consider renaming `environmentConfig` to something like `pipelineEnvironment` or `environmentForPipeline`?